analyzers/nodejs/nodejs.go

```diff
@@ -91,10 +86,8 @@ func New(m module.Module) (*Analyzer, error) {
 		NodeCmd:     nodeCmd,
 		NodeVersion: nodeVersion,
 
-		Tool: npmTool,
-
-		YarnCmd:     yarnCmd,
-		YarnVersion: yarnVersion,
+		NPMTool:  npmTool,
+		YarnTool: yarn.New(),
```

Comment:
We might want to move the `os.Getenv` checking logic into here, unless anything else uses `yarn.New()` or `npm.New()`.